subscription-engine/kafka-ws/app/Redis/RedisManager.js

```diff
@@ -21,9 +22,39 @@ class RedisManager {
     // TODO: Implement "retry_strategy"
     this.redisClient = redis.createClient(this.config);
     logger.info('RedisManager singleton creation complete!');
+
+    const stateService = 'redis';
+    StateManager.registerService(stateService);
+    StateManager.registerShutdownHandler(this.shutdownHandler.bind(this));
+    this.redisClient.on('connect', () => StateManager.signalReady(stateService));
+    /**
+     * The 'error' event must be mapped, otherwise the application hangs on an uncaughtException
+     * and some unexpected behaviors happens.
+     *
+     * The error event doesn't mean the service is unhealthy, because it can be
+     * AbortError, ParserError AggregateError, or others subclasses of RedisError
+     * When the client disconnects to redis the 'end' event is fired, there we can consider
+     * the service is unhealthy
+     */
+    this.redisClient.on('error', (error) => logger.warn(`${error}`));
+    this.redisClient.on('end', () => StateManager.signalNotReady(stateService));
+
     return Object.seal(this);
   }
 
+  /**
+   * Shutdown handler to be passed to the ServiceStateManager.
+   *
+   * @returns {Promise<void>}
+   *
+   * @function shutdown Handler
+   * @public
+   */
+  shutdownHandler() {
+    logger.warn('Shutting redis, disconnecting...');
+    return this.redisClient.end(true);
+  }
```

Comment:
Note that we are doing the **graceful** shutdown of the service, but this `redisClient.end(true)` method forces the request to close:
> Forcibly close the connection to the Redis server. Note that this does not wait until all replies have been parsed. If you want to exit cleanly, call client.quit() as mentioned above.

Instead of forcing, use `redisClient.quit(callback)`: https://www.npmjs.com/package/redis#clientquitcallback

**Note that** you need to return a `Promise` that will only be _resolved_ when a `callback` is called, this is necessary to ensure that the `shutdownHandler()` performs his role completely.